### PR TITLE
Refactor: Define PodMetricsClient interface and hide implementation details of vllm metrics processing

### PR DESCRIPTION
**Pull request by liu-cong**

### CODE REVIEW

File: pkg/ext-proc/backend/provider.go [hunk 1]
```diff
@@ -130,3 +130,37 @@ func (p *Provider) refreshPodsOnce() error {
 	p.podMetrics.Range(mergeFn)
 	return nil
 }
+
+func (p *Provider) refreshMetricsOnce() error {
```
Comment (liu-cong) [inline pkg/ext-proc/backend/provider.go:134]: This is moved from metrics.go, no new code

File: pkg/ext-proc/backend/vllm/metrics.go [hunk 2]
```diff
@@ -25,45 +28,38 @@ const (
...
-				multierr.Append(errs, fmt.Errorf("failed to get all pod metrics updated from prometheus: %v", err))
-			}
-			p.UpdatePodMetrics(pod, updated)
-		}()
-		return true
+type PodMetricsClientImpl struct {
+}
+
+// FetchMetrics fetches metrics from a given pod.
+func (p *PodMetricsClientImpl) FetchMetrics(pod backend.Pod, existing *backend.PodMetrics) (*backend.PodMetrics, error) {
```
Comment (liu-cong) [inline pkg/ext-proc/backend/vllm/metrics.go:35]: This is moved from pod_client.go, no new code

File: pkg/ext-proc/main.go [hunk 3]
```diff
@@ -75,7 +73,7 @@ func main() {
 
 	s := grpc.NewServer()
 
-	pp := backend.NewProvider(&backend.PodMetricsClientImpl{}, &backend.FakePodLister{Pods: pods})
+	pp := backend.NewProvider(&vllm.PodMetricsClientImpl{}, &backend.FakePodLister{Pods: pods})
```
Comment (ahg-g) [inline pkg/ext-proc/main.go:76]: why did we move it to a vllm package? shouldn't this be agnostic the model server?

File: pkg/ext-proc/backend/provider.go (hunk 1 above)
Comment (kfswain) [reply pkg/ext-proc/backend/provider.go:134]: The name of this file is slightly confusing. It reads as if it's intended to be a factory class, but then we have metrics related funcs, while there is a metrics.go file. 

What is this file intended to do?

File: pkg/ext-proc/main.go (hunk 3 above)
Comment (liu-cong) [reply pkg/ext-proc/main.go:76]: Model servers don't always share the same metric names, so I expect we will need some "adapter code" for each model server.

File: pkg/ext-proc/backend/provider.go (hunk 1 above)
Comment (liu-cong) [reply pkg/ext-proc/backend/provider.go:134]: yeah probably the name is too broad. It intends to "provide info of the backend", thus "provider", and metrics is part of the "info".

File: pkg/ext-proc/main.go (hunk 3 above)
Comment (liu-cong) [reply pkg/ext-proc/main.go:76]: We will likely share some helper functions across model servers, but until we integrate with the next model server, I put metrics scraping code in vllm.

**Comment by ahg-g** [comment]
/lgtm
/approve

just so we move forward

**Pull request merged**